ffi/src/lib.rs

```diff
@@ -432,13 +432,16 @@ unsafe fn unwrap_and_parse_path_as_url(path: KernelStringSlice) -> DeltaResult<U
 /// A builder that allows setting options on the `Engine` before actually building it
 #[cfg(any(feature = "default-engine", feature = "sync-engine"))]
 pub struct EngineBuilder {
+    #[cfg(feature = "default-engine")]
     url: Url,
     allocate_fn: AllocateErrorFn,
+    #[cfg(feature = "default-engine")]
     options: HashMap<String, String>,
 }
 
 #[cfg(any(feature = "default-engine", feature = "sync-engine"))]
 impl EngineBuilder {
+    #[cfg(feature = "default-engine")]
```

Comment:
If we don't allow setting options on the sync engine, there's probably no point using the builder at all in that case? Just `get_sync_engine` is probably enough?

Reply:
Good point!

Reply:
I was code-monkeying late at night and did not consider this